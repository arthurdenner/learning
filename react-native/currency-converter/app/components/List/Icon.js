import React from 'react';
import PropTypes from 'prop-types';
import { Image, View } from 'react-native';
import styles from './styles';

const Icon = ({ checkmark, visible }) => {
  const iconStyles = [styles.icon];

  if (visible) {
    iconStyles.push(styles.iconVisible);
  }

  return (
    <View style={iconStyles}>
      {checkmark && (
        <Image
          resizeMode="contain"
          source={require('./images/check.png')}
          style={styles.checkIcon}
        />
      )}
    </View>
  );
};

Icon.propTypes = {
  checkmark: PropTypes.bool,
  visible: PropTypes.bool,
};

Icon.defaultProps = {
  checkmark: true,
  visible: true,
};

export default Icon;
