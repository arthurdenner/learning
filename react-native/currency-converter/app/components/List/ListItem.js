import React from 'react';
import PropTypes from 'prop-types';
import { Text, TouchableHighlight, View } from 'react-native';
import Icon from './Icon';

import styles from './styles';

const ListItem = ({ onPress, selected, text, ...rest }) => (
  <TouchableHighlight underlayColor={styles.$underlayColor} onPress={onPress}>
    <View style={styles.row}>
      <Text style={styles.text}>{text}</Text>
      {selected && <Icon {...rest} />}
    </View>
  </TouchableHighlight>
);

ListItem.propTypes = {
  onPress: PropTypes.func.isRequired,
  selected: PropTypes.bool,
  text: PropTypes.string.isRequired,
};

ListItem.defaultProps = {
  selected: false,
};

export default ListItem;
