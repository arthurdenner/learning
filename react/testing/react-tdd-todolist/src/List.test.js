import React from 'react';
import List from './List';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('list component', () => {
  it('should render a todo item', () => {
    const tree = toJson(shallow(<List items={[]} />));

    expect(tree).toMatchSnapshot();
  });

  it('should match its snapshot with items', () => {
    const items = ['Learn react', 'rest', 'go out'];
    const tree = renderer.create(<List items={items} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
