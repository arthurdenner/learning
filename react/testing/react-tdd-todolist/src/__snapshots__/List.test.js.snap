// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list component should match its snapshot with items 1`] = `
<ul>
  <li>
    Learn react
  </li>
  <li>
    rest
  </li>
  <li>
    go out
  </li>
</ul>
`;

exports[`list component should render a todo item 1`] = `<ul />`;
